util/submitter_amqp.go

```diff
@@ -202,11 +202,26 @@ func (s *AMQPSubmitter) SubmitWithHeaders(rawData []byte, key string, contentTyp
 		option["headers"].(origamqp.Table)[k] = v
 	}
 
+	s.Submitter.ConnMutex.Lock()
+	// Double-check whether a value is nil and make sure we won't deref it.
+	if s.Submitter == nil {
```

Comment:
Just a comment, because I do like the switch thing in Go somehow:

```
switch {
case s.Submitter == nil:
     ...
     return
case s.Submitter.Channel == nil:
     ...
     return
}
```

It is the same, but it is nice grouping those (then it only needs one comment indeed.

Reply:
:+1: